### PR TITLE
pacman: Switch to meson

### PR DESCRIPTION
**Pull request by elieux**
Spent a bit of time on the meson build.  Currently I have this issue:

```
Checking for function "ngettext" : NO
```

I'll see if I can get further tomorrow.

### CODE REVIEW

**Comment by lazka** [comment]
`-Di18n=false` should work around that for now

**Comment by elieux** [comment]
No large issues remaining.  Only these:

- libalpm.pc now contains:
    /usr/lib/gcc/x86_64-pc-msys/9.3.0/../../../../lib/libintl.a
- almost every test fails with something like:
    TAP parsing error: unexpected input at line (1,)

Merge?

**Comment by eli-schwartz** [comment]
The ngettext thing should be fixed by the patch I mentioned at https://github.com/msys2/MSYS2-packages/pull/1549#issuecomment-505671734

Current version of the patch is here: https://git.archlinux.org/users/eschwartz/pacman.git/commit/?h=queue2&id=b297114e4daa141944a639ab11b8bca77af39b8e if I can get confirmation it works as expected I'll try to get it in soon.

**Comment by lazka** [comment]
This is what I get for ldd:
```
$ ldd pkg/pacman/usr/bin/pacman.exe
        ntdll.dll => /c/WINDOWS/SYSTEM32/ntdll.dll (0x7ffbf0800000)
        KERNEL32.DLL => /c/WINDOWS/System32/KERNEL32.DLL (0x7ffbeeda0000)
        KERNELBASE.dll => /c/WINDOWS/System32/KERNELBASE.dll (0x7ffbed9b0000)
        ADVAPI32.dll => /c/WINDOWS/System32/ADVAPI32.dll (0x7ffbeec80000)
        msvcrt.dll => /c/WINDOWS/System32/msvcrt.dll (0x7ffbf00c0000)
        msys-gpgme-11.dll => /usr/bin/msys-gpgme-11.dll (0x5d3120000)
        msys-2.0.dll => /usr/bin/msys-2.0.dll (0x180040000)
        sechost.dll => /c/WINDOWS/System32/sechost.dll (0x7ffbefc10000)
        RPCRT4.dll => /c/WINDOWS/System32/RPCRT4.dll (0x7ffbefd20000)
        msys-gpg-error-0.dll => /usr/bin/msys-gpg-error-0.dll (0x45a000000)
        msys-assuan-0.dll => /usr/bin/msys-assuan-0.dll (0x5a3090000)
        msys-gcc_s-seh-1.dll => /usr/bin/msys-gcc_s-seh-1.dll (0x5e8160000)
        msys-intl-8.dll => /usr/bin/msys-intl-8.dll (0x430b30000)
        msys-iconv-2.dll => /usr/bin/msys-iconv-2.dll (0x5603f0000)
````

expected?

File: pacman/PKGBUILD [hunk 1]
```diff
@@ -119,7 +132,13 @@ prepare() {
...
+  patch -p1 -i ${srcdir}/0013-use-libintl.patch
+  patch -p1 -i ${srcdir}/0014-fix-asciidoc-argparse.patch
+  patch -p1 -i ${srcdir}/0015-excise-ldconfig.patch
+  patch -p1 -i ${srcdir}/0016-excise-fakeroot.patch
+  patch -p1 -i ${srcdir}/0017-excise-sudo.patch
+  patch -p1 -i ${srcdir}/0018-use-msys-tools.patch
+
   cp ${srcdir}/Doxyfile.in doc/
 
   autoreconf -fi
```
Comment (lazka) [inline pacman/PKGBUILD:144]: Is this still needed?

**Comment by lazka** [comment]
(Some of the patches still contain hunks for autotools which could be dropped)

**Comment by elieux** [comment]
> Current version of the patch is here: https://git.archlinux.org/users/eschwartz/pacman.git/commit/?h=queue2&id=b297114e4daa141944a639ab11b8bca77af39b8e if I can get confirmation it works as expected I'll try to get it in soon.

I have it planned for tomorrow.

> (Some of the patches still contain hunks for autotools which could be dropped)

I know.  I assumed we'd deal with it once autotools is dropped from upstream.

File: pacman/PKGBUILD (hunk 1 above)
Comment (elieux) [reply pacman/PKGBUILD:144]: Probably not.

**Comment by elieux** [comment]
The `ldd` result is definitely not expected. How did you build?

**Comment by lazka** [comment]
> The `ldd` result is definitely not expected. How did you build?

just makepkg with this PR. Only difference is that I use a newer meson, I'll try downgrading.

**Comment by lazka** [comment]
> The `ldd` result is definitely not expected. How did you build?

Same with older meson. Could it be I need 9a5045442d34 ?

**Comment by lazka** [comment]
> Same with older meson. Could it be I need [9a50454](https://github.com/msys2/MSYS2-packages/commit/9a5045442d347df75228d4d1a7b0e8804497dd90) ?

yes, that's it

**Comment by lazka** [comment]
>     * almost every test fails with something like:
>       TAP parsing error: unexpected input at line (1,)

```patch
--- pacman-5.2.1/test/pacman/pmtest.py.orig	2019-10-24 15:29:08.000000000 +0200
+++ pacman-5.2.1/test/pacman/pmtest.py	2020-05-18 09:28:26.343194400 +0200
@@ -287,8 +268,10 @@
         # Change to the tmp dir before running pacman, so that local package
         # archives are made available more easily.
         time_start = time.time()
-        self.retcode = subprocess.call(cmd, stdout=output, stderr=output,
-                cwd=os.path.join(self.root, util.TMPDIR), env={'LC_ALL': 'C'})
+        self.retcode = subprocess.call(cmd,
+                cwd=os.path.join(self.root, util.TMPDIR),
+                env={'LC_ALL': 'C', 'PATH': os.environ['PATH']},
+                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
         time_end = time.time()
         vprint("\ttime elapsed: %.2fs" % (time_end - time_start))
 
```

**Comment by elieux** [comment]
@lazka: Sorry about gpgme, I didn't realize it's not built yet. Thanks for the test fix, it seems to be back to normal.

@eli-schwartz: I replaced my patch with yours, seems to be fine.

**Comment by Alexpux** [comment]
> 
> 
> Builds successfully. No large issues remaining. Only these:
> 
>     * libalpm.pc now contains:
>       /usr/lib/gcc/x86_64-pc-msys/9.3.0/../../../../lib/libintl.a
> 
>     * almost every test fails with something like:
>       TAP parsing error: unexpected input at line (1,)
> 
> 
> Merge?

This leads to not possible rebuild pacman

**Pull request merged**